server/routes/overview/prisonerSummaryRequestHandler.ts

```diff
@@ -0,0 +1,35 @@
+import type { PrisonerSummary } from 'viewModels'
+import createError from 'http-errors'
+import { NextFunction, Request, RequestHandler, Response } from 'express'
+import PrisonerSearchService from '../../services/prisonerSearchService'
+
+/**
+ * Class exposing middleware functions related to ensuring we have the correct PrisonerSummary in session.
+ */
+export default class PrisonerSummaryRequestHandler {
+  constructor(private readonly prisonerSearchService: PrisonerSearchService) {}
+
+  /**
+   *  Middleware function to look up the prisoner from prisoner-search, map to a PrisonerSummary, and store in the session
+   */
+  getPrisonerSummary: RequestHandler = async (req: Request, res: Response, next: NextFunction) => {
```

Comment:
This function is pretty much what we originally had in `OverviewController`
Implementing it here as a `RequestHandler` means it can be reused as part of the middleware stack for the routes 👍